services/subscription_service.go

```diff
@@ -39,6 +39,22 @@ func (s *SubcriptionService) CreateSubscription(ctx context.Context, groupID str
 		return nil, NewServiceError(http.StatusBadRequest, err)
 	}
 
+	app, err := s.appRepo.FindApplicationByID(ctx, newSubscription.AppID)
```

Comment:
We should add a similar check for the source ID too if it's present in the request.

Reply:
I mean, why not here as well? @danvixent

Reply:
Nothing really. I'll do it here